### PR TITLE
chore: Remove model hub

### PR DESCRIPTION
**Pull request by MikhailKardash**
## Ticket
https://hpe-aiatscale.atlassian.net/browse/MD-461

## Description
Remove `model_hub` library and subdirectory from `determined`

## Test Plan
No testing required

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .circleci/real_config.yml [hunk 1]
```diff
@@ -5632,7 +5498,7 @@ workflows:
           name: publish-python-package-rc
           matrix:
             parameters:
-              path: ["harness", "model_hub"]
+              path: ["harness"]
```
Comment (dannysauer) [inline .circleci/real_config.yml:5501]: I wish people would just use the regular YAML lists for consistency :)
```suggestion
              path:
                - "harness"
```

File: .circleci/real_config.yml [hunk 2]
```diff
@@ -5649,7 +5515,7 @@ workflows:
           name: publish-python-package-release
           matrix:
             parameters:
-              path: ["harness", "model_hub"]
+              path: ["harness"]
```
Comment (dannysauer) [inline .circleci/real_config.yml:5518]:
```suggestion
              path:
                - "harness"
```

File: docs/.redirects/redirects.json [hunk 3]
```diff
@@ -14,8 +20,8 @@
     "architecture/system-architecture": "../get-started/architecture/system-architecture.html",
     "architecture/introduction": "../get-started/architecture/introduction.html",
     "setup-cluster/deploy-cluster/slurm/install-on-slurm": "../../slurm/install-on-slurm.html",
-    "setup-cluster/deploy-cluster/slurm/hpc-with-agent": "../../slurm/_index.html", 
-    "setup-cluster/slurm/hpc-with-agent": "../slurm/_index.html",
+    "setup-cluster/deploy-cluster/slurm/hpc-with-agent": "../../slurm/_index.html",
+    "setup-cluster/slurm/hpc-with-agent": "_index.html",
```
Comment (dannysauer) [inline docs/.redirects/redirects.json:24]: Not related, right? ;)

File: docs/.redirects/redirects.json [hunk 4]
```diff
@@ -466,4 +472,4 @@
     "tutorials/porting-tutorial": "pytorch-mnist-tutorial.html",
     "tutorials/quick-start": "quickstart-mdldev.html",
     "tutorials/pytorch-mnist-local-qs": "../get-started/webui-qs.html"
-}
\ No newline at end of file
+}
```
Comment (dannysauer) [inline docs/.redirects/redirects.json:475]: Thank you for adding missing newlines back.

/Grumbles at whatever web editor some people use which keeps pruning one byte from files.

**Review by dannysauer — APPROVED** [review]
LGTM.  A couple of optional grumbles to prove I read the whole PR.

File: docs/.redirects/redirects.json (hunk 3 above)
Comment (MikhailKardash) [reply docs/.redirects/redirects.json:24]: I had to fix some redirects using `docs/redirects.py`. This is auto-generated.

**Review by tara-det-ai — APPROVED** [review]
LGTM

**Review by jgongd — COMMENTED** [review]
Other places that have model hub we may want to remove:
1. The model hub section in `.github/dependabot.yml`.
2. Install model_hub section in `.github/lint-python.yml`.

**Review by jgongd — APPROVED** [review]
LGTM

**Comment by jgongd** [comment]
@azhou-determined! Could you take a look at this too? I know you've done a ton of work on the agent side.

**Review by azhou-determined — APPROVED** [review]
lgtm.

i approved and ran the e2e-longrunning tests. there's a few failures but they don't seem related to this PR, though i didn't dig too deeply.

**Comment by mackrorysd** [comment]
Failures all disappeared on retry, and are being looked into elsewhere as flaky. The failure that remains is the one with the ChromaDB version. But these e2e test runs aren't free, so I'm not going to rebase and re-run - I think we've established that this patch passes tests. Time to merge!

**Pull request merged**